### PR TITLE
feat: add migrations and initial endpoints for user lists

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds database migrations for user lists
- Adds CRUD endpoints for managing user lists

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Related to: https://github.com/open-sauced/engineering/issues/107

## Mobile & Desktop Screenshots/Recordings

image

Sample output

```json
{
  "id": "uuid-v4",
  "user_id": 237133,
  "name": "JavaScript Developers",
  "is_public": false,
  "created_at": "2022-10-19 13:24:51.000000",
  "updated_at": "2022-10-19 13:24:51.000000",
  "deleted_at": "2022-10-19 13:24:51.000000"
}
```

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: migrations/20230905145301_add_user_list_contributors.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+create table if not exists public.user_list_contributors
+(
+  -- static columns
+  id uuid default uuid_generate_v4() not null,
+  list_id uuid not null references public.user_lists (id) on delete cascade on update cascade,
```
Comment (bdougie) [inline migrations/20230905145301_add_user_list_contributors.sql:5]: does this cover the case where someone is on multiple lists? or are the UserListContributors all unique and there will be multiple of these objects for based on how many lists I am on?

**Review by bdougie — COMMENTED** [review]
Just have a question on the data structure for users on lists and how they will related.

File: migrations/20230905145301_add_user_list_contributors.sql (hunk 1 above)
Comment (brandonroberts) [reply migrations/20230905145301_add_user_list_contributors.sql:5]: Yes, someone can be on multiple lists with this structure, as all user list contributors have their own record associated to a list ID. The cascade on the delete only applies to the contributors with that list ID.

**Review by bdougie — APPROVED** [review]

File: src/user-lists/entities/user-list-contributor.entity.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+} from "typeorm";
+import { ApiHideProperty } from "@nestjs/swagger";
+import {
+  ApiModelProperty,
+  ApiModelPropertyOptional,
+} from "@nestjs/swagger/dist/decorators/api-model-property.decorator";
+
+import { DbUser } from "../../user/user.entity";
+
+@Entity({ name: "user_organizations" })
```
Comment (jpmcb) [inline src/user-lists/entities/user-list-contributor.entity.ts:18]: Can you hit that `npm run generate:swagger` 🙈 we really need to come up with a hook to do this for us. Maybe we can have that as part of a "pre-release" step when things merge to `main`.

**Review by jpmcb — APPROVED** [review]
Looks great!!!! 👏🏼

File: src/user-lists/entities/user-list-contributor.entity.ts (hunk 2 above)
Comment (brandonroberts) [reply src/user-lists/entities/user-list-contributor.entity.ts:18]: "I'm gonna generate the swagger doc this time" 

Got me again 😅

**Review by jpmcb — APPROVED** [review]

**Pull request merged**